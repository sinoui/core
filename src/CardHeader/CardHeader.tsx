import React from 'react';
import styled from 'styled-components';

export interface Props {
  /**
   * 卡片头部头像
   */
  avatar?: React.ReactNode;
  /**
   * 卡片头部操作
   */
  action?: React.ReactNode;
  /**
   * 卡片头部主标题
   */
  title: string;
  /**
   * 卡片头部副标题
   */
  subheader?: string;
}
const CardHeaderWrapper = styled.div`
  display: flex;
  padding: 0 16px;
  height: 72px;
  align-items: center;
  box-sizing: border-box;
  font-size: ${(props) => props.theme.typography.body2.fontSize};
  & .card-header-avatar {
    flex: 0 0 auto;
    margin-right: 16px;
  }

  & .card-header-content {
    flex: 1 1 auto;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  & .card-header-title {
    color: ${(props) => props.theme.palette.text.primary};
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  & .card-header-subheader {
    color: ${(props) => props.theme.palette.text.secondary};
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  & .card-header-action {
    flex: 0 0 auto;
  }
`;

/**
 * CardHeader 没有头像和action时使用 props.theme.typography.h6
 */
const CardHeaderTitle = styled.div((props) => props.theme.typography.h6);

/**
 *
 * CardHeader  卡片头部组件
 */
export default function CardHeader({
  avatar,
  action,
  title,
  subheader,
}: Props) {
  return (
    <CardHeaderWrapper>
      {avatar && <div className="card-header-avatar">{avatar}</div>}
      <div className="card-header-content">
        {avatar && action ? (
          <div className="card-header-title">{title}</div>
        ) : (
          <CardHeaderTitle>{title}</CardHeaderTitle>
        )}

        {subheader && <div className="card-header-subheader">{subheader}</div>}
      </div>
      {action && <div className="card-header-action">{action}</div>}
    </CardHeaderWrapper>
  );
}
