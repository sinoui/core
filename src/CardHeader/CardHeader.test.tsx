import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import '@testing-library/jest-dom/extend-expect';
import CardHeader from './CardHeader';

const Avatar = () => {
  return (
    <div
      style={{
        height: '40px',
        width: '40px',
        backgroundColor: '#f44336',
        borderRadius: '4px',
      }}
    />
  );
};

it('正确渲染CardHeader', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <CardHeader
          avatar={Avatar}
          title="Shrimp and Chorizo Paella"
          subheader="subheader"
        />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
