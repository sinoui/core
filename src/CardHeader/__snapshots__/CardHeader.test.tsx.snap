// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 只有主标题的CardHeader 1`] = `
<div
  className="sc-AxjAm ioqRtj"
>
  <div
    className="card-header-content"
  >
    <div
      className="sc-AxirZ hBVueb"
    >
      Shrimp and Chorizo Paella
    </div>
  </div>
</div>
`;

exports[`快照测试 头像+主标题+副标题+操作的CardHeader 1`] = `
<div
  className="sc-AxjAm ioqRtj"
>
  <div
    className="card-header-avatar"
  />
  <div
    className="card-header-content"
  >
    <div
      className="card-header-title"
    >
      Shrimp and Chorizo Paella
    </div>
    <div
      className="card-header-subheader"
    >
      subheader
    </div>
  </div>
  <div
    className="card-header-action"
  >
    <span>
      icon
    </span>
  </div>
</div>
`;
